### PR TITLE
drakrun: changed qemu-img to qemu-img-xen

### LINKED ISSUE #152: Use qemu-img-xen instead of qemu-img
Currently, we're requiring qemu-utils as a dependency which we can avoid by simply using qemu-img-xen provided by Xen hypervisor

### PR DESCRIPTION
**Pull request by manorit2001**
Fix #152

### CODE REVIEW

**Review by icedevml — CHANGES_REQUESTED** [review]
```
Mar 11 16:13:50 zen drakrun[5329]: Traceback (most recent call last):
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 215, in check_tools
Mar 11 16:13:50 zen drakrun[5329]:     subprocess.check_output("qemu-img-xen --version", shell=True)
Mar 11 16:13:50 zen drakrun[5329]:   File "/usr/lib/python3.7/subprocess.py", line 395, in check_output
Mar 11 16:13:50 zen drakrun[5329]:     **kwargs).stdout
Mar 11 16:13:50 zen drakrun[5329]:   File "/usr/lib/python3.7/subprocess.py", line 487, in run
Mar 11 16:13:50 zen drakrun[5329]:     output=stdout, stderr=stderr)
Mar 11 16:13:50 zen drakrun[5329]: subprocess.CalledProcessError: Command 'qemu-img-xen --version' returned non-zero exit status 1.
Mar 11 16:13:50 zen drakrun[5329]: During handling of the above exception, another exception occurred:
Mar 11 16:13:50 zen drakrun[5329]: Traceback (most recent call last):
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/bin/drakrun", line 5, in <module>
Mar 11 16:13:50 zen drakrun[5329]:     dm.cmdline_main()
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/main.py", line 608, in cmdline_main
Mar 11 16:13:50 zen drakrun[5329]:     main(args)
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/main.py", line 653, in main
Mar 11 16:13:50 zen drakrun[5329]:     c.init_drakrun()
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/main.py", line 178, in init_drakrun
Mar 11 16:13:50 zen drakrun[5329]:     generate_vm_conf(self.install_info, self.instance_id)
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/vm.py", line 26, in generate_vm_conf
Mar 11 16:13:50 zen drakrun[5329]:     storage_backend = get_storage_backend(install_info)
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 327, in get_storage_backend
Mar 11 16:13:50 zen drakrun[5329]:     return REGISTERED_BACKENDS[install_info.storage_backend](install_info)
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 209, in __init__
Mar 11 16:13:50 zen drakrun[5329]:     self.check_tools()
Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 217, in check_tools
Mar 11 16:13:50 zen drakrun[5329]:     raise RuntimeError("Failed to determine qemu-img version. "
Mar 11 16:13:50 zen drakrun[5329]: RuntimeError: Failed to determine qemu-img version. Make sure you have qemu-utils installed.
Mar 11 16:13:50 zen systemd[1]: drakrun@1.service: Main process exited, code=exited, status=1/FAILURE
```

This is due to the fact that `qemu-img-xen` shipped with Xen is from year 2008 o_O and doesn't support `--version` switch

**Comment by icedevml** [comment]
@chivay Do we still want to downgrade to `qemu-img-xen` in such situation? If so, the version check would have to be adjusted somehow.

**Comment by manorit2001** [comment]
> ```
> Mar 11 16:13:50 zen drakrun[5329]: Traceback (most recent call last):
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 215, in check_tools
> Mar 11 16:13:50 zen drakrun[5329]:     subprocess.check_output("qemu-img-xen --version", shell=True)
> Mar 11 16:13:50 zen drakrun[5329]:   File "/usr/lib/python3.7/subprocess.py", line 395, in check_output
> Mar 11 16:13:50 zen drakrun[5329]:     **kwargs).stdout
> Mar 11 16:13:50 zen drakrun[5329]:   File "/usr/lib/python3.7/subprocess.py", line 487, in run
> Mar 11 16:13:50 zen drakrun[5329]:     output=stdout, stderr=stderr)
> Mar 11 16:13:50 zen drakrun[5329]: subprocess.CalledProcessError: Command 'qemu-img-xen --version' returned non-zero exit status 1.
> Mar 11 16:13:50 zen drakrun[5329]: During handling of the above exception, another exception occurred:
> Mar 11 16:13:50 zen drakrun[5329]: Traceback (most recent call last):
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/bin/drakrun", line 5, in <module>
> Mar 11 16:13:50 zen drakrun[5329]:     dm.cmdline_main()
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/main.py", line 608, in cmdline_main
> Mar 11 16:13:50 zen drakrun[5329]:     main(args)
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/main.py", line 653, in main
> Mar 11 16:13:50 zen drakrun[5329]:     c.init_drakrun()
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/main.py", line 178, in init_drakrun
> Mar 11 16:13:50 zen drakrun[5329]:     generate_vm_conf(self.install_info, self.instance_id)
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/vm.py", line 26, in generate_vm_conf
> Mar 11 16:13:50 zen drakrun[5329]:     storage_backend = get_storage_backend(install_info)
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 327, in get_storage_backend
> Mar 11 16:13:50 zen drakrun[5329]:     return REGISTERED_BACKENDS[install_info.storage_backend](install_info)
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 209, in __init__
> Mar 11 16:13:50 zen drakrun[5329]:     self.check_tools()
> Mar 11 16:13:50 zen drakrun[5329]:   File "/opt/venvs/drakrun/lib/python3.7/site-packages/drakrun/storage.py", line 217, in check_tools
> Mar 11 16:13:50 zen drakrun[5329]:     raise RuntimeError("Failed to determine qemu-img version. "
> Mar 11 16:13:50 zen drakrun[5329]: RuntimeError: Failed to determine qemu-img version. Make sure you have qemu-utils installed.
> Mar 11 16:13:50 zen systemd[1]: drakrun@1.service: Main process exited, code=exited, status=1/FAILURE
> ```
> 
> This is due to the fact that `qemu-img-xen` shipped with Xen is from year 2008 o_O and doesn't support `--version` switch

Oops

**Comment by manorit2001** [comment]
> @chivay Do we still want to downgrade to `qemu-img-xen` in such situation? If so, the version check would have to be adjusted somehow.

We can use `which qemu-img-xen` in that case maybe, as it doesn't seem like we are using the version-specific checks anyway

File: drakrun/drakrun/storage.py [hunk 1]
```diff
@@ -251,16 +250,15 @@ def rollback_vm_storage(self, vm_id: int):
         except FileNotFoundError:
             pass
 
+        # dummy commit
```
Comment (manorit2001) [inline drakrun/drakrun/storage.py:253]: Just trying out this feature :p
```suggestion
```

**Comment by chivay** [comment]
OK, I wasn't expecting that Xen does this thing. I guess that using normal qemu-utils doesn't hurt.
My initial thought would be that we'd be able to reduce the number of deb package dependencies, however
if it's so problematic I think that we should leave it as it is.

**Comment by icedevml** [comment]
Closing as "wontfix", sorry for that.

**Pull request closed without merging**